src/components/queue/QueueList/QueueList.tsx

```diff
@@ -1,89 +1,159 @@
-import React, {FC, useEffect, useState} from "react";
-import {
-    Box,
-    CircularProgress,
-    Grid,
-    Stack,
-    Typography,
-} from "@mui/material";
-import Button from "@components/shared/Button";
+
+import BouncingCubesAnimation from "@components/animations/BouncingCubesAnimation";
 import CreateTicketDialog from "@components/queue/CreateTicketDialog";
 import QueueListItem from "@components/queue/QueueListItem";
-import {useTickets} from "@util/queue/hooks";
-import {Queue, Ticket} from "@util/queue/api";
-import {useAuth} from "@util/auth/hooks";
-import BouncingCubesAnimation from "@components/animations/BouncingCubesAnimation";
+import Button from "@components/shared/Button";
+import {
+  Box,
+  Chip,
+  CircularProgress,
+  Divider,
+  Grid,
+  Stack,
+  Tooltip,
+  Typography
+} from "@mui/material";
+import { useAuth } from "@util/auth/hooks";
+import { Queue, Ticket, TicketStatus } from "@util/queue/api";
+import { useTickets } from "@util/queue/hooks";
+import { TICKET_COOLDOWN_MINUTES } from "@util/shared/constants";
+import React, { FC, useEffect, useState } from "react";
 
 export interface QueueListProps {
-    queue: Queue
-    showCompletedTickets: boolean;
+  queue: Queue;
+  showCompletedTickets: boolean;
 }
 
 /**
  * QueueList lists out the tickets in a queue.
  */
 const QueueList: FC<QueueListProps> = ({queue, showCompletedTickets}) => {
     const {currentUser, isTA} = useAuth();
-    const [tickets, ticketsLoading] = useTickets(queue.id, showCompletedTickets);
+    const [tickets, ticketsLoading] = useTickets(queue.id, false);
+    const shownTickets = tickets?.filter(t => !showCompletedTickets || t.status !== TicketStatus.StatusComplete);
     const [createTicketDialog, setCreateTicketDialog] = useState(false);
 
-    const inQueue = tickets && tickets.filter(ticket => ticket.user.Email == currentUser?.email).length > 0;
+    const [currentTime, setCurrentTime] = useState(new Date());
+
+    useEffect(() => {
+        const id = setInterval(() => {
+            setCurrentTime(new Date());
+        }, 5000);
+        return () => clearInterval(id);
+    });
```

Comment:
why update current time every 5 seconds?

Reply:
This is a question for @nickyy96

Reply:
I believe this was just the time we decided was reasonable to update. @alexander-ding what are your thoughts on this?

Reply:
Refreshing the current time might trigger a rerender. The ticket submit button is enabled when currentTime - lastTicketTime is greater than 5 minutes. This doesn't need to happen too frequently, but we can lower the interval to something like one second. Shouldn't be a big deal either way since React shouldn't rerender unless we cross the 5 minute threshold.

Reply:
I believe that a render will be triggered every time you call the set state function (every 5 seconds) -- if the state variable was instead a boolean that determined whether the button should be disabled or not, we could avoid calling the set state as often (only when it is actually necessary to update the button) and prevent unnecessary renders.

Reply:
I think React may be smart enough to not trigger DOM updates if the actual states directly invoked in what's returned hasn't changed? But definitely don't quote me on this. We can do the refactor just to be safe

Reply:
It appears that React re-renders the entire component whenever the set state function is called, per this Codepen https://codepen.io/lbain/pen/MJNpwL?editors=1111

Reply:
This should be resolved in 03bd382aeb798452a6e29fcfd8b056fdfa65a664